editor/osm_auth.cpp

```diff
@@ -20,18 +22,20 @@ constexpr char const * kApiVersion = "/api/0.6";
 namespace
 {
 
-string FindAuthenticityToken(string const & body)
+string FindAuthenticityToken(std::string const & action, string const & body)
 {
-  auto pos = body.find("name=\"authenticity_token\"");
-  if (pos == string::npos)
-    return {};
-  string const kValue = "value=\"";
-  auto start = body.find(kValue, pos);
-  if (start == string::npos)
-    return {};
-  start += kValue.length();
-  auto const end = body.find('"', start);
-  return end == string::npos ? string() : body.substr(start, end - start);
+  static std::regex const kActionAndTokenRE(R"~(action="(.+?)".*?name="authenticity_token" value="(.+?)")~");
+
+  auto begin = std::sregex_iterator{body.begin(), body.end(), kActionAndTokenRE};
```

Comment:
Since begin is not const, I suppose you mean this:
```
std::sregex_iterator end{};
for (auto it = std::sregex_iterator{body.begin(), body.end(), kActionAndTokenRE}; it != end; ++it)
```

Reply:
Nope, I meant `const` :)